File: NYTViewer-project/app/src/main/java/com/gmail/dp/denzay/nytviewer/MainActivity.java
package com.gmail.dp.denzay.nytviewer;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.gmail.dp.denzay.nytviewer.adapters.FavouritesDBAdapter;
import com.gmail.dp.denzay.nytviewer.adapters.SectionsPagerAdapter;
import com.gmail.dp.denzay.nytviewer.models.NewsItem;
import com.gmail.dp.denzay.nytviewer.views.FavouritesActivity;
import com.gmail.dp.denzay.nytviewer.views.OnListFragmentInteractionListener;
import com.gmail.dp.denzay.nytviewer.views.WebViewActivity;

public class MainActivity extends AppCompatActivity implements OnListFragmentInteractionListener {

    private SectionsPagerAdapter mSectionsPagerAdapter;
    private FavouritesDBAdapter mDBAdapter;
    private ViewPager mViewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mDBAdapter = FavouritesDBAdapter.getInstance();
        mDBAdapter.connect(getApplicationContext());

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        // Create the adapter that will return a fragment for each of the three
        // primary sections of the activity.
        mSectionsPagerAdapter = new SectionsPagerAdapter(getSupportFragmentManager());

        // Set up the ViewPager with the sections adapter.
        mViewPager = (ViewPager) findViewById(R.id.container);
        mViewPager.setAdapter(mSectionsPagerAdapter);

        TabLayout tabLayout = (TabLayout) findViewById(R.id.tabs);

        mViewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
        tabLayout.addOnTabSelectedListener(new TabLayout.ViewPagerOnTabSelectedListener(mViewPager));
    }

    @Override
    protected void onDestroy() {
        mDBAdapter.disconnect();
        super.onDestroy();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.action_view_favourites) {
            showFavouritesActivity();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onListFragmentInteraction(NewsItem item) {
        Intent intent = new Intent(this, WebViewActivity.class);
        intent.putExtra(WebViewActivity.KEY_NEWS_ITEM, item);
        intent.putExtra(WebViewActivity.KEY_IS_CACHED_ITEM,  false);
        startActivity(intent);
    }

    private void showFavouritesActivity() {
        Intent intent = new Intent(this, FavouritesActivity.class);
        startActivity(intent);
    }

}

File: NYTViewerMVVM/app/src/main/java/com/gmail/dp/denzay/nytviewer_mvvm/MainActivity.kt
package com.gmail.dp.denzay.nytviewer_mvvm

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.gmail.dp.denzay.nytviewer_mvvm.presentation.activity_main.news.common.NewsFragmentPageAdapter
import com.gmail.dp.denzay.nytviewer_mvvm.presentation.common.MainRouter
import com.gmail.dp.denzay.nytviewer_mvvm.presentation.common.base.BaseActivity
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject

class MainActivity : BaseActivity() {

    private var mSectionsPagerAdapter: NewsFragmentPageAdapter? = null

    @Inject
    lateinit var router: MainRouter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setSupportActionBar(toolbar)
        mSectionsPagerAdapter = NewsFragmentPageAdapter(supportFragmentManager)
        container.adapter = mSectionsPagerAdapter
    }


    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        if (id == R.id.action_view_favourites) {
            router.openFavourites()
            return true
        }

        return super.onOptionsItemSelected(item)
    }
}
